src/modules/proxy/adapters/custom-websocket-adapter.ts

```diff
@@ -0,0 +1,165 @@
+import * as WebSocket from "ws";
+import * as SocketIO from "socket.io";
+import { WebSocketAdapter } from "@nestjs/common";
+import { MessageMappingProperties } from "@nestjs/websockets";
+import { Observable, fromEvent, EMPTY, fromEventPattern } from "rxjs";
+import { mergeMap, filter } from "rxjs/operators";
+
+type HybridWebsocketServer = WebSocket.Server | SocketIO.Server;
+type HybridWebsocketClient = WebSocket.WebSocket | SocketIO.Socket;
+
+export enum SocketServerType {
+  SOCKET_IO = "/socket.io",
+  WEBSOCKET = "/ws",
+  AI_ASSISTANT = "/ai-assistant",
+}
+
+export interface SocketServerOptions {
+  type: SocketServerType;
+  path?: string;
+  cors?: {
+    origin?: string;
+  };
+}
+
+export class CustomWebSocketAdapter implements WebSocketAdapter {
+  constructor(private app: any) {}
```

Comment:
@LordNayan  sir if we have to use different scoket io in future so will it be possible to add them in this custom adapter?

Reply:
Yes, it is possible. We will just need to add a new path enum in SocketServerType enum and add it to the switch case in create function.